pwnlib/tubes/ssh.py

```diff
@@ -1031,7 +1048,29 @@ def is_exe(path):
 %(func_src)s
 %(func_name)s(*%(func_args)r)
 
-os.execve(exe, argv, env)
+# os.execve does not allow us to pass empty argv[0]
+# Therefore we use ctypes to call execve directly
+# os.execve(exe, argv, env)
+        
+# Transform envp from dict to list
+env_list = [key + b"=" + value for key, value in env.items()]
+
+# ctypes helper to convert a python list to a NULL-terminated C array
+def to_carray(py_list):
+    py_list += [None] # NULL-terminated
+    return (ctypes.c_char_p * len(py_list))(*py_list)
+
+c_argv = to_carray(argv)
+c_env = to_carray(env_list)
+
+# Call execve
+libc = ctypes.CDLL(None)
```

Comment:
Do you have a reference of where `ctypes.CDLL` accepts `None` as `name` to get libc? Above we're using `ctypes.CDLL('libc.so.6')`.